core/src/impl/Kokkos_Serial_Task.hpp

```diff
@@ -76,14 +76,18 @@ class TaskQueueSpecialization<SimpleTaskScheduler<Kokkos::Serial, QueueType> > {
   static void execute(scheduler_type const& scheduler) {
     using task_base_type = typename scheduler_type::task_base_type;
 
+    execution_space serial_execution_space;
```

Comment:
That does not look right.
Shouldn't it be
```C++
auto const& serial_execution_space = scheduler.get_execution_space();

Reply:
No it's correct.

Reply:
Would you explain to me?

Reply:
The object has a single data member, a pointer to a singleton. The two codes are equivalent.

Reply:
But if we relax that then this is a bug isn't it?

Reply:
I don't know what you want me to say about this. Yes hypothetical change to the code may or may not require other changes.